web/packages/design/src/Modal/Modal.tsx

```diff
@@ -50,26 +119,39 @@ export default class Modal extends React.Component {
     }
   }
 
+  dialogEl = (): Element => {
```

Comment:
This is going to be quite brittle, as it depends on the rendering logic in a way that is unusual to a typical React component. (I'm not even sure how it would work if called in different parts of the component lifecycle, like when `props` have already been updated, but not yet reflected in the DOM). I would consider using a simple `div` that wraps only the "dialog" element itself, and then `ref`'ing this one. This way, we can always access the element we need as `firstElementChild`. Unless, of course, that would interfere with something else.

Reply:
I'm inclined to keep it. It depends on reading a ref, so I'd say it's as brittle as using refs in general. It does not help that this is a class component, so technically you have all bunch of different lifecycle methods that are no longer supported, as you mentioned.

If you look at the current use, `dialogEl` is called from `componendDidMount`, `componentDidUpdate` and event handlers, [all of which are called after a render happens](https://legacy.reactjs.org/docs/react-component.html#the-component-lifecycle), so the ref is guaranteed to exist.

I'd rather avoid adding a div because it's a breaking change if there's code that renders a modal that depends on the DOM having a certain hierarchy.